src/pool/srv_pool.c

```diff
@@ -2267,6 +2267,7 @@ pool_prop_read(struct rdb_tx *tx, const struct pool_svc *svc, uint64_t bits,
 			obj_ver = 0;
 			prop->dpp_entries[idx].dpe_flags |= DAOS_PROP_ENTRY_NOT_SET;
 		} else if (rc != 0) {
+			daos_prop_free(prop);
```

Comment:
Looks like you could also
```suggestion
			D_GOTO(out_prop, rc);
```